rust-runtime/aws-smithy-runtime/src/client/retries/classifiers.rs

```diff
@@ -119,41 +167,78 @@ impl HttpStatusCodeClassifier {
             retryable_status_codes: retryable_status_codes.into(),
         }
     }
+
+    /// Return the priority of this retry classifier.
+    pub fn priority() -> RetryClassifierPriority {
+        RetryClassifierPriority::HttpStatusCodeClassifier
+    }
+}
+
+impl From<HttpStatusCodeClassifier> for SharedRetryClassifier {
+    fn from(value: HttpStatusCodeClassifier) -> Self {
+        Self::new(value)
+    }
 }
 
 impl ClassifyRetry for HttpStatusCodeClassifier {
-    fn classify_retry(&self, ctx: &InterceptorContext) -> Option<RetryReason> {
+    fn classify_retry(
+        &self,
+        ctx: &InterceptorContext,
+        _: Option<RetryClassifierResult>,
+    ) -> Option<RetryClassifierResult> {
         ctx.response()
             .map(|res| res.status().as_u16())
             .map(|status| self.retryable_status_codes.contains(&status))
             .unwrap_or_default()
-            .then_some(RetryReason::Error(ErrorKind::TransientError))
+            .then_some(RetryClassifierResult::Error(ErrorKind::TransientError))
     }
 
     fn name(&self) -> &'static str {
         "HTTP Status Code"
     }
+
+    fn priority(&self) -> RetryClassifierPriority {
+        Self::priority()
+    }
 }
 
-// Generic smithy clients would have something like this:
-// pub fn default_retry_classifiers() -> RetryClassifiers {
-//     RetryClassifiers::new()
-//         .with_classifier(SmithyErrorClassifier::new())
-//         .with_classifier(ModeledAsRetryableClassifier::new())
-//         .with_classifier(HttpStatusCodeClassifier::new())
-// }
-// This ordering is different than the default AWS ordering because the old generic client classifier
-// was the same.
+/// Given an iterator of retry classifiers and an interceptor context, run retry classifiers on the
+/// context. Each classifier is passed the classification result from the previous classifier (the
+/// 'root' classifier is passed `None`.)
+pub fn run_classifiers_on_ctx(
+    classifiers: impl Iterator<Item = SharedRetryClassifier>,
+    ctx: &InterceptorContext,
+) -> RetryClassifierResult {
+    let mut result = None;
+
+    for classifier in classifiers {
+        result = classifier.classify_retry(ctx, result);
+        match result.as_ref() {
+            Some(result) => tracing::debug!(
+                "running retry classifier '{}' returned result '{result}'",
+                classifier.name()
+            ),
+            None => tracing::debug!(
+                "running retry classifier '{}' returned no result",
+                classifier.name()
+            ),
```

Comment:
this is _probably_ too noisy at the debug level because we're going to get multiple log lines for every request. Maybe only log if a retry is actually indicated?